src/Faker/Provider/Image.php

```diff
@@ -28,41 +35,55 @@ class Image extends Base
      *
      * @return string
      */
-    public static function imageUrl($width = 640, $height = 480, $category = null, $randomize = true, $word = null, $gray = false)
-    {
-        $baseUrl = "https://lorempixel.com/";
-        $url = "{$width}/{$height}/";
+    public static function imageUrl(
+        $width = 640,
+        $height = 480,
+        $category = null,
+        $randomize = true,
+        $word = null,
+        $gray = false
+    ) {
+        $size = sprintf('%dx%d.png', $width, $height);
 
-        if ($gray) {
-            $url = "gray/" . $url;
+        $imageParts = [];
+        if ($category !== null) {
+            $imageParts[] = $category;
         }
-
-        if ($category) {
-            if (!in_array($category, static::$categories)) {
-                throw new \InvalidArgumentException(sprintf('Unknown image category "%s"', $category));
-            }
-            $url .= "{$category}/";
-            if ($word) {
-                $url .= "{$word}/";
-            }
+        if ($word !== null) {
+            $imageParts[] = $word;
         }
-
-        if ($randomize) {
-            $url .= '?' . static::randomNumber(5, true);
+        if ($randomize === true) {
+            $imageParts[] = Lorem::word();
         }
 
-        return $baseUrl . $url;
+        $backgroundColor = $gray === true ? 'CCCCCC' : str_replace('#', '', Color::safeHexColor());
+
+        return sprintf(
+            '%s/%s/%s%s',
+            self::BASE_URL,
+            $size,
+            $backgroundColor,
+            count($imageParts) > 0 ? '?text=' . urlencode(implode(' ', $imageParts)) : null
```

Comment:
In case there are no `$imageParts`, `null` will be converted to string. It works, but making it an empty string instead would be even better :)
```suggestion
            count($imageParts) > 0 ? '?text=' . urlencode(implode(' ', $imageParts)) : ''
```